src/main/kotlin/ecommerce/service/UserService.kt

```diff
@@ -0,0 +1,37 @@
+package ecommerce.service
+
+import ecommerce.dto.LoggedInUser
+import ecommerce.entity.User
+import ecommerce.repository.UserRepository
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Service
+import org.springframework.web.server.ResponseStatusException
+
+@Service
+class UserService(
+    private val userRepository: UserRepository,
+) {
+    fun register(
+        email: String,
+        password: String,
+        role: String?,
+    ): Long {
+        if (userRepository.existsByEmail(email)) throw ResponseStatusException(HttpStatus.BAD_REQUEST, "Email already in use.")
+
+        val user = User(email = email, password = password, role = role ?: "USER")
+        return userRepository.create(user)
+    }
```

Comment:
I think `User` constructor could have default role parameter.
and it's not a just string. we could declare enum class.